### PR TITLE
Fix filesystem testcase issues

### PR DESCRIPTION
**Pull request by lokeshbv**
- Fix blocking of fs testcase
- Fix failure of tc_fs_vfs_stat()
- Move procfs tc's from fs_main to tc_procfs_main
- Fix wrong usage of configs wrapper
- Modify mounting un-mounting of procfs
- Fix fs readdir() tc failure

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [46b22f0116e97af21a7d28c4b979181343a91ef9] - Code Rule Check Result:
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:115: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:124: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:211: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:249: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:275: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('

**Comment by seinfra** [comment]
Target : [46b22f0116e97af21a7d28c4b979181343a91ef9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d121dab55bb2f21a5733206dfefc86d68b9525af] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d121dab55bb2f21a5733206dfefc86d68b9525af] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 1]
```diff
@@ -3556,11 +3553,27 @@ int tc_filesystem_main(int argc, char *argv[])
 
 	tc_fs_vfs_rename();
 	tc_fs_vfs_ioctl();
+
+#ifndef CONFIG_FS_AUTOMOUNT_PROCFS
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: I think this condition is needed in tc_fs_procfs_main instead of mouting of procfs here.

Comment (jeongarmy) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: I know that you changed these because of stat of procfs in tc_fs_vfs_stat().
I think it is better to keep tc_fs_vfs_stat() and add condition about procfs to "ret = stat(PROCFS_PATH, &st);" in tc_fs_vfs_stat.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 2]
```diff
@@ -265,20 +266,15 @@ void tc_fs_smartfs_procfs_main(void)
...
-	}
 
 	ret = read_dir_entries(PROC_MOUNTPOINT);
 	TC_ASSERT_EQ("read_dir_entries", ret, OK);
 
 	ret = procfs_rewind_tc(PROC_MOUNTPOINT);
 	TC_ASSERT_EQ("procfs_rewind_tc", ret, OK);
 
-#if defined(CONFIG_FS_PROCFS) && !defined (CONFIG_FS_PROCFS_EXCLUDE_UPTIME)
+#if defined(CONFIG_FS_PROCFS) && !defined(CONFIG_FS_PROCFS_EXCLUDE_UPTIME)
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: please remove unused variable

File: apps/examples/testcase/tc_common.h [hunk 3]
```diff
@@ -139,6 +139,7 @@ extern int total_fail;
 {\
 	printf("\n[%s] PASS \n", __func__); \
 	total_pass++; \
+	return; \
```
Comment (jeongarmy) [inline apps/examples/testcase/tc_common.h]: please exclude changes which are not related to this commit

Comment (lokeshbv) [reply apps/examples/testcase/tc_common.h]: this is the fix for one of the fs tc blocking issue

**Comment by seinfra** [comment]
Target : [3c135ccdbd9c4623df4768671aa116a3e8afd7e2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3c135ccdbd9c4623df4768671aa116a3e8afd7e2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0b41582f079a4e95d0e07ca8701333195cb6848a] - Code Rule Check Result:
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:115: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:124: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:211: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c:249: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

**Comment by seinfra** [comment]
Target : [0b41582f079a4e95d0e07ca8701333195cb6848a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d4850be926fcd60ed9697fc33c9191180f27d6e4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d4850be926fcd60ed9697fc33c9191180f27d6e4] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 4]
```diff
@@ -953,7 +953,7 @@ static void tc_fs_vfs_seekdir(void)
 	seekdir(dir, offset);
 	TC_ASSERT_NEQ_CLEANUP("seekdir", dir, NULL, closedir(dir));
 	dirent = readdir(dir);
-	TC_ASSERT_NEQ_CLEANUP("readdir", dirent, NULL, closedir(dir));
+	TC_ASSERT_EQ_CLEANUP("readdir", dirent, NULL, closedir(dir));
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Why this changed?

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 5]
```diff
@@ -34,15 +34,16 @@
 #include <sys/stat.h>
 #include "tc_common.h"
 
-#define PROC_MOUNTPOINT "/proc"
+#define PROCFS_TEST_MOUNTPOINT "/test_proc"
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: no, we gonna use /proc like a linux

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 6]
```diff
@@ -3330,33 +3327,7 @@ static void tc_libc_stdio_zeroinstream(void)
...
-* @postcondition    NA
-*/
-#if !defined(CONFIG_FS_PROCFS_EXCLUDE_MTD)
-static void tc_driver_mtd_procfs_ops(void)
-{
-	int fd;
-	int ret;
-	struct stat st;
 
-	fd = open(MTD_PROCFS_PATH, O_RDONLY);
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Please clean constant values PROCFS_PATH / MTD_PROCFS_PATH,

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 7]
```diff
@@ -53,8 +54,8 @@
 #define SMARTFS_DEV_PATH "/dev/smart1"
 #endif
 
-#define PROC_SMARTFS_PATH "/proc/fs/smartfs"
-#define PROC_SMARTFS_FILE_PATH "/proc/fs/smartfs/file.txt"
+#define PROC_SMARTFS_PATH "/test_proc/fs/smartfs"
```
[the remark this answers is not in the document]
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: May be missed, I will modify it

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 4 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Please see commit description

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 6 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Ok

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c (hunk 5 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: But it is good to do like this. So that there will not be any dependency on "/proc" mount point used by kernel

File: apps/examples/testcase/tc_common.h (hunk 3 above)
Comment (lokeshbv) [reply apps/examples/testcase/tc_common.h]: I know that issue is due to double freeing of pointer.
Fixes are,
1. Assign NULL to pointer after first free.
2. Prevent fall through due to TC_SUCCESS_RESULT macro which intend leads to double freeing of pointer.
3. Fix tc application is blocking due to double free, which is due to some semaphore issues ,  I guess.

Reason for selecting Fix 2 is ,  while writing this tc it is assumed that TC_SUCCESS_RESULT will return, which actually right. Addition to this fix we can also add Fix 1 but in this case it doesn't have logical effect. And Fix 3 needs time so that we can check it later

**Comment by lokeshbv** [comment]
@pradeepns85 , yes there are some typo mistakes in commits. I will modify those.

**Comment by seinfra** [comment]
Target : [3c611dac855647aced3e55b252bcc2ac10b43701] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3c611dac855647aced3e55b252bcc2ac10b43701] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c61b09919370e1b0b780cb81c3fc01a363e03d95] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c61b09919370e1b0b780cb81c3fc01a363e03d95] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [034c0e39c247d26e0d927289efdf1834fe667f65] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [034c0e39c247d26e0d927289efdf1834fe667f65] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cb144632d305c45edb5cace811c6a7cee1e44274] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cb144632d305c45edb5cace811c6a7cee1e44274] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 8]
```diff
@@ -33,16 +33,20 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include "tc_common.h"
+#ifndef CONFIG_FS_AUTOMOUNT_PROCFS
+#include <stdbool.h>
+#endif
 
-#define PROC_MOUNTPOINT "/proc"
+#define PROCFS_TEST_MOUNTPOINT "/proc"
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: How about "/test_proc" or "/proc_test"? "/proc" is a too common name so that it is easily duplicated with user's proc mount point.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 9]
```diff
@@ -53,8 +57,10 @@
 #define SMARTFS_DEV_PATH "/dev/smart1"
 #endif
 
-#define PROC_SMARTFS_PATH "/proc/fs/smartfs"
-#define PROC_SMARTFS_FILE_PATH "/proc/fs/smartfs/file.txt"
+#define PROC_SMARTFS_PATH PROCFS_TEST_MOUNTPOINT"/fs/smartfs"
+#define PROC_SMARTFS_FILE_PATH PROCFS_TEST_MOUNTPOINT"/fs/smartfs/file.txt"
+
+#ifdef CONFIG_TC_FS_PROCFS
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: This file is already in same conditional on Make.defs. This is redundant.

**Comment by sunghan-chang** [comment]
@lokeshbv We want to leave a title as what you modify in your commit. ```Fix blocking of fs testcase``` is a result of that change. How about "add return at end of TC_SUCCESS_RESULT macro```?

**Comment by seinfra** [comment]
Target : [c4fe390707d8306a6792b5a44f7cf81fa0f9d29a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c4fe390707d8306a6792b5a44f7cf81fa0f9d29a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [03a68667844e63b56a5b92b7336bf97806a12d4e] -

**Comment by seinfra** [comment]
Target : [03a68667844e63b56a5b92b7336bf97806a12d4e] -

**Comment by seinfra** [comment]
Target : [d104b546acf160bb0542ebe4fc3d76c3c1617e7b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d104b546acf160bb0542ebe4fc3d76c3c1617e7b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [025cd752928a44adb72ad093ce0c65a7e1bfd6e2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [025cd752928a44adb72ad093ce0c65a7e1bfd6e2] - Code Rule Check OK.
